lib/rodauth/features/base.rb

```diff
@@ -470,7 +472,22 @@ def database_function_password_match?(name, hash_id, password, salt)
     end
 
     def password_hash_match?(hash, password)
-      BCrypt::Password.new(hash) == password
+      case get_hash_algorithm(hash)
+      when 'bcrypt'
+        BCrypt::Password.new(hash) == password
+      when 'argon2'
+        Argon2::Password.verify_password(password, hash)
+      end
```

Comment:
It might be worth catching an error here when the hash is invalid and thinking about a more general error. For example, when the user disabled bcrypt and accounts with the old hash remained in the database and ended up in the branch with argon2.